src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -151,3 +152,62 @@ def signature_hash_algorithm(self):
             raise UnsupportedAlgorithm(
                 "Signature algorithm OID:{0} not recognized".format(oid)
             )
+
+    @property
+    def extensions(self):
+        # TODO: remove this when we finish landing support for extensions.
+        warnings.warn(
+            "Extension support is not fully implemented. Unsupported "
+            "extensions are IGNORED at this time."
+        )
+        extensions = []
+        seen_oids = set()
+        extcount = self._backend._lib.X509_get_ext_count(self._x509)
+        for i in range(0, extcount):
+            ext = self._backend._lib.X509_get_ext(self._x509, i)
+            assert ext != self._backend._ffi.NULL
+            crit = self._backend._lib.X509_EXTENSION_get_critical(ext)
+            critical = crit == 1
+            nid = self._backend._lib.OBJ_obj2nid(ext.object)
+            assert nid != self._backend._lib.NID_undef
+            if nid == self._backend._lib.NID_basic_constraints:
+                value = self._build_basic_constraints(ext)
+                oid = x509.OID_BASIC_CONSTRAINTS
```

Comment:
I don't have any preference. The NID approach avoids some additional memory allocations (all basic constraints objects share the same ObjectIdentifier object) but that's a very minor thing. I'm happy to change it.